### PR TITLE
Add warn about requirements.txt file to "requirements without setup.py" section of readme

### LINKED ISSUE #744: Readme: "requirements without setup.py" section doesn't warn that it considers existing requirements.txt file
I just've lost couple of hours trying to figure out why `pip-compile` resolves my `requirements.in` to some old versions of packages. I'd like to propose adding a note to "requirements without setup.py" readme section, that would mention how to migrate from "plain `requirements.txt` workflow" to "`requirements.in` with pip-tools" without confusion.

##### Steps to replicate

1. Have requirements.txt with some package pinned.
2. Follow Readme in Example usage of `pip-compile` -> Without `setup.py`.

##### Expected result

Readme in this section warns the reader that his existing `requirements.txt` might interfere.

##### Actual result

Readme doesn't warn and reader is confused with the result of pip-compile.

### PR DESCRIPTION
**Pull request by hramezani**
Resolves #744.

**Changelog-friendly one-liner**: Add warn about requirements.txt file to "requirements without setup.py" section of readme.

##### Contributor checklist

- ~~Provided the tests for the changes.~~
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Comment by atugushev** [comment]
Hello @tysonclugg,

Sorry for bothering you. I've requested a review from you since you are the author of the changed paragraph. Could you please review it when you have time? Thanks!

File: README.rst [hunk 1]
```diff
@@ -88,7 +88,9 @@ Without ``setup.py``
 --------------------
 
 If you don't use ``setup.py`` (`it's easy to write one`_), you can create a
-``requirements.in`` file to declare the Flask dependency:
+``requirements.in`` **(make sure you don't have a requirement.txt file,
+otherwise, it might interfere with existing requirements.txt)** file to declare
```
Comment (atugushev) [inline README.rst]: Could the repetitive word `requirements.txt` be avoided?

Comment (hramezani) [reply README.rst]: `...otherwise, it might interfere.` would be good?

**Comment by tysonclugg** [comment]
The warning is reasonable, but it applies equally if building `requirements.txt` from either `setup.py` or `requirements.in`.  The best solution may be to include a similar warning in a opening statement about what `pip-sync` and other tools are supposed to achieve, rather than repeating the same warning in multiple sections.

**Comment by hramezani** [comment]
@tysonclugg Thanks,
@atugushev I moved it as a note to the top of the readme file.

**Comment by atugushev** [comment]
@tysonclugg good point! Thanks :)

**Comment by hramezani** [comment]
@atugushev it's done.

**Comment by atugushev** [comment]
@mbeacom, thanks for reviewing this!

**Pull request merged**